doc/proposals/ansible-operator-testing.md

```diff
@@ -0,0 +1,36 @@
+# Ansible Operator Testing
+
+## Background
+All operators should fit into the e2e testing framework used by operator-sdk, including Ansible Operator.
+
+## Goals
+
+- Add support for the `test {local|cluster}` e2e testing subcommand with Ansible Operators
+- Integrate with existing Ansible test frameworks to provide a uniform experience across the Ansible ecosystem
+- Maintain the existing interface for the `test` subcommands
+
+## Non-Goals
+
+- Create a new testing framework for Ansible that plays nicely with operators
+- Allow Ansible users to write non-Ansible tests
+
+## Solution
+1. Update scaffolding to (optionally?) include [molecule](https://molecule.readthedocs.io/en/latest/) initialization
+    - Set up a molecule scenario for the e2e environment (may have different behavior between `local` and `cluster` scenarios)
+1. Create a [delegated driver](https://molecule.readthedocs.io/en/latest/configuration.html#delegated) for molecule that:
```

Comment:
I was thinking about that too, but I'm not sure if I'd be able to maintain something that spins up a kubernetes environment for testing... It would be pretty cool though.

Reply:
I suppose we could have it spin up a minikube instance or something if there isn't already one available, but it might be more trouble than its worth regardless.

Reply:
Although it would be generically useful for developing roles for Kubernetes as well, not just specific to operators.

Reply:
I pretty much went with this approach